### PR TITLE
Allow setting pod security context through CRD

### PR DESCRIPTION
**Pull request by saketmht**
This implements #373 

Changes:

1. Added `PodSecurityContext` and `SecurityContext` fields for cluster and dashboard pods.
2. Added unit test for the same.
3. Doc updates for the added fields.

### CODE REVIEW

**Comment by saketmht** [comment]
@swoehrl-mw Can you help me with why the lint job failed? I don't see any error messaged in the run and local lint check didn't report any errors either. And could you review the PR as well?

**Comment by swoehrl-mw** [comment]
@saketmht Looks like the linter job ran into out-of-memory. Not much we can do except try to fiddle with the configuration. But if it passes locally then that's good enough for me.
Regarding a review: It'll likely be a few days until I can get to it, if @idanl21 pitches in it might be faster.

File: opensearch-operator/pkg/builders/cluster.go [hunk 1]
```diff
@@ -406,10 +406,11 @@ func NewSTSForNodePool(
...
 							},
-							StartupProbe:   &probe,
-							LivenessProbe:  &probe,
-							ReadinessProbe: &readinessProbe,
-							VolumeMounts:   volumeMounts,
+							StartupProbe:    &probe,
+							LivenessProbe:   &probe,
+							ReadinessProbe:  &readinessProbe,
+							VolumeMounts:    volumeMounts,
+							SecurityContext: cr.Spec.General.SecurityContext,
```
Comment (idanl21) [inline opensearch-operator/pkg/builders/cluster.go:413]: Lets add parameter for that , to keep on the convention .

File: opensearch-operator/pkg/builders/dashboards.go [hunk 2]
```diff
@@ -151,17 +151,19 @@ func NewDashboardsDeploymentForCR(cr *opsterv1.OpenSearchCluster, volumes []core
...
-							LivenessProbe: &probe,
-							Env:           env,
-							VolumeMounts:  volumeMounts,
-							Command:       mainCommand,
+							StartupProbe:    &probe,
+							LivenessProbe:   &probe,
+							Env:             env,
+							VolumeMounts:    volumeMounts,
+							Command:         mainCommand,
+							SecurityContext: cr.Spec.Dashboards.SecurityContext,
```
Comment (idanl21) [inline opensearch-operator/pkg/builders/dashboards.go:159]: also here

**Review by idanl21 — APPROVED** [review]
Hey, just lets add parmetrs on builders, also cloned it, and LINT passed so it sok for my side

**Comment by saketmht** [comment]
@swoehrl-mw @idanl21 Updated PR according to comments.

**Review by swoehrl-mw — CHANGES_REQUESTED** [review]
@saketmht Can you add a little section to the userguide about this feature, maybe also explain in one sentence why someone should need this? That should make it easier to find for users.
Code LGTM.

**Comment by saketmht** [comment]
@swoehrl-mw Added a section for security context in the userguide.

**Review by swoehrl-mw — APPROVED** [review]

**Pull request merged**